src/d3a/models/area/__init__.py

```diff
@@ -199,9 +199,10 @@ def tick(self, is_root_area=False):
                 for market in self.all_markets:
                     market.match_offers_bids()
         self.events.update_events(self.now)
-        if self.current_tick % self.config.ticks_per_slot == 0 and is_root_area:
-            self._cycle_markets()
-        self.dispatcher.broadcast_tick()
+        if not ConstSettings.GeneralSettings.DISPATCH_EVENTS_BOTTOM_TO_TOP:
```

Comment:
I would like to try to use this code for both cases (bottom-to-top and top-to-bottom). Therefore, I would like to see if it is feasible to first cycle all markets in both cases, and then start dispatching the tick events. If you decide to try and do this, the code related to cycling markets in the tick event will be removed, and the broadcast_tick call for both cases will reside in the event_listener. If it is not clear let's discuss this offline.

Reply:
Sorry, just saw your comment @spyrostz .
Ok, I will try that. But wasn't it part of the story, that we get rid of the connection between tick and market cycle events?
So NOT doing this at all?

Reply:
Sorry, just understood what you meant. Trying it.